api/repositories/org_repository.go

```diff
@@ -393,6 +401,91 @@ func (r *OrgRepo) GetSpace(ctx context.Context, info authorization.Info, spaceGU
 	return spaceRecords[0], nil
 }
 
+func (r *OrgRepo) DeleteOrg(ctx context.Context, info authorization.Info, message DeleteOrgMessage) error {
+	// Fetch 'hierarchy' object using the privileged client as we want to return a 'not-found' error for
+	// non-existent orgs as opposed to saying 'permission-denied'.
+	hierarchyObj := v1alpha2.HierarchyConfiguration{}
+	err := r.privilegedClient.Get(ctx, types.NamespacedName{Name: "hierarchy", Namespace: message.GUID}, &hierarchyObj)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return NotFoundError{
+				Err: err,
+			}
+		}
+		return err
+	}
+
+	if r.authEnabled {
```

Comment:
I'd restructure this. Instead of having two separate, almost-identical paths, we can branch to choose which client to use, then rejoin a single path to do the updates and deletes. The only difference I see between the two paths is the check for IsForbidden, and I don't think it's an issue to have that for both cases.

Reply:
The caveat I would add here is that we plan to remove one of the two almost-identical paths soon, as we don't really want to support the non RBAC use case; so we should make sure that it will be easy to see what code ought to be discarded when we go to discard it.

Having said that, I don't think that doing the refactor @matt-royal suggests would cause problems--we'd just be shrinking the conditional logic down to a single block where we set the client.

Reply:
I have tried it keep it simple and easy to remove later - changes in commit https://github.com/cloudfoundry/cf-k8s-controllers/pull/464/commits/a99f88f188236f227f4a094857f8ada66c9204e7